### ISSUE TITLE
Is this plugin functional? 

### ISSUE THREAD

**Issue by danmincu** [samsta/quarantine#11]
Hello I built the plugin, installed it in Jenkins 2.138.3, setup a freestyle project that points to a test Maven project with couple of JUnit tests. 
One of the tests is purposely setup to _fail_ and after running the build once I quarantined it. 
The test quarantined status is retained, however the job fails. 
I wonder why do people report this plugin as **working** since the whole purpose of quarantining a test is to have the build succeeding?!

Attached is the log, please note in the end how `0 unquarantined failures remaining`

```
Building in workspace E:\GitHub\quaratineTests
[quaratineTests] $ cmd.exe /C "C:\JavaTools\apache-maven\bin\mvn.cmd install && exit %%ERRORLEVEL%%"
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building quaratineTests 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ quaratineTests ---
[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 0 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ quaratineTests ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ quaratineTests ---
[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory E:\GitHub\quaratineTests\src\test\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ quaratineTests ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ quaratineTests ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running FooServiceTests
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.04 s <<< FAILURE! - in FooServiceTests
[ERROR] testFooFooMethod(FooServiceTests)  Time elapsed: 0.004 s  <<< FAILURE!
java.lang.AssertionError: expected:<30> but was:<4>
	at FooServiceTests.testFooFooMethod(FooServiceTests.java:20)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   FooServiceTests.testFooFooMethod:20 expected:<30> but was:<4>
[INFO] 
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.341 s
[INFO] Finished at: 2018-11-21T15:12:51-05:00
[INFO] Final Memory: 22M/981M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project quaratineTests: There are test failures.
[ERROR] 
[ERROR] Please refer to E:\GitHub\quaratineTests\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Build step 'Invoke top-level Maven targets' marked build as failure
Recording test results
[Quarantine]: failed sending email: javax.mail.MessagingException: Could not connect to SMTP host: localhost, port: 25;
  nested exception is:
	java.net.ConnectException: Connection refused: connect
[Quarantine]: FooServiceTests.testFooFooMethod failed but is quarantined
[Quarantine]: 0 unquarantined failures remaining
Finished: FAILURE
```

**Comment by samsta** [comment]
That's because the build has already failed before the quarantine plugin gets to run. Every build step can either keep the build result the same or make it worse, it can't make it better. So once it is FAILURE, the plugin can't make it go SUCCESS.

Therefore, you should only exit with an error if there's an actual compilation error or some other problem with the build. A failing test should not make you build return with an error and leave it up to the JUnit plugin to decide whether or not it is SUCCESS or UNSTABLE.

In short:

ERROR: compilation etc. error
UNSTABLE: test failures
SUCCESS: none of the above

**Comment by danmincu** [comment]
Great, let me just double check with you the steps one more time. So one should build a pipeline that 
1) runs compile + code analysis - return the appropriate status
2) run unit + integration tests but return *success* regardless of test failures, it publishes the results in the JUnit format.
3) let the JUNIT plugin parse those results and set the final job status, however, the plugin will be ignoring the outcome of the quarantined test.
Let me know if I missed something,
Thanks!!

**Comment by samsta** [comment]
That should work, yes. You can do the compile and test in a single job if you want to. I see that you're working in Windows which I have no experience with, but on Linux you can do something like

`mvn compile
mvn test || true`

which allows the compile to fail your job, but the test always passes, and the JUnit/quarantine plugin then makes the decision based on the XML file
